pontoon/base/models/entity.py

```diff
@@ -843,24 +845,60 @@ def for_project_locale(
         if search:
             search_list = utils.get_search_phrases(search)
 
-            translation_filters = (
-                Q(translation__string__icontains_collate=(search, locale.db_collation))
-                & Q(translation__locale=locale)
-                for search in search_list
-            )
-            translation_matches = entities.filter(*translation_filters).values_list(
-                "id", flat=True
-            )
+            if search_identifiers:
+                translation_filters = (
+                    Q(
+                        translation__string__icontains_collate=(
+                            search,
+                            locale.db_collation,
+                        )
+                    )
+                    & Q(translation__locale=locale)
+                    for search in search_list
+                )
+                translation_matches = entities.filter(*translation_filters).values_list(
+                    "id", flat=True
+                )
+            else:
+                entities = entities.annotate(
+                    filtered_string=Substr("string", StrIndex("string", Value("=")) + 1)
```

Comment:
This is only relevant for Fluent strings. Not sure if adding a condition for this will make the query faster or slower.

Reply:
Also, maybe we should call it `string_without_identifier` or `string_without_id` rather than `filtered_string`?

Reply:
Wait, shouldn't we here be annotating `filtered_string` (or whatever we call it) to translations rather than entities?